test/unit/spec/EnrollTotpController_spec.js

```diff
@@ -689,319 +530,600 @@ function (Okta, OktaAuth, LoginUtil, Util, DeviceTypeForm, BarcodeForm,
+        })
+          .then(function (test) {
+            Expect.isVisible(test.passCodeForm.form());
+            test.passCodeForm.setCode(123456);
+            test.setNextResponse(resSuccess);
+            test.passCodeForm.submit();
+            expect($.ajax.calls.count()).toBe(1);
+            Expect.isJsonPost($.ajax.calls.argsFor(0), {
+              url: 'https://foo.com/api/v1/authn/factors/id1234/lifecycle/activate',
+              data: {
+                passCode: '123456',
+                stateToken: expectedStateToken
+              }
             });
+          });
+      });
+      itp('returns to factor list when browser\'s back button is clicked', function () {
+        return setupOktaTotpFn({}, true)
+          .then(function (test) {
+            return enrollFactor(test, totpEnrollSuccessRes);
+          })
+          .then(function (test) {
+            return Expect.waitForBarcodeTotp(test);
+          })
+          .then(function (test) {
+            test.scanCodeForm.submit();
+            return Expect.waitForActivateTotp(test);
+          })
+          .then(function (test) {
+            test.setNextResponse(resAllFactors);
+            Util.triggerBrowserBackButton();
+            return Expect.waitForEnrollChoices(test);
+          })
+          .then(function (test) {
+            Expect.isEnrollChoices(test.router.controller);
+            Util.stopRouter();
+          });
+      });
+
+    }
+
+    Expect.describe('Header & Footer', function () {
+      itp('displays the correct factorBeacon for Okta Verify', function () {
+        return setupOktaTotp().then(function (test) {
+          expect(test.beacon.isFactorBeacon()).toBe(true);
+          expect(test.beacon.hasClass('mfa-okta-verify')).toBe(true);
+        });
+      });
+      itp('displays the correct factorBeacon for Google Authenticator', function () {
+        return setupGoogleTotp().then(function (test) {
+          expect(test.beacon.isFactorBeacon()).toBe(true);
+          expect(test.beacon.hasClass('mfa-google-auth')).toBe(true);
+        });
+      });
+      itp('has a "back" link in the footer', function () {
+        return setupOktaTotp().then(function (test) {
+          Expect.isVisible(test.form.backLink());
         });
       });
     });
 
+    Expect.describe('Enroll factor', function () {
+      testEnrollFactor(setupOktaTotp, setupAndEnrollOktaTotp, setupGoogleTotp, setupAndEnrollGoogleTotp, 'testStateToken');
+    });
+
+    Expect.describe('Enroll factor on Idx Pipeline', function () {
+      testEnrollFactor(setupOktaTotpWithIdxPipeline, setupAndEnrollOktaTotpWithIdxPipeline, setupGoogleTotpWithIdxPipeline, setupAndEnrollGoogleTotpWithIdxPipeline, '01testStateToken');
+    });
+
+    Expect.describe('Scan qrcode', function () {
+      testScanQRCode(setupAndEnrollOktaTotp, setupOktaTotp, resTotpEnrollSuccess, resAllFactors);
+      Expect.describe('Scan qrcode on Idx Pipeline', function () {
```

Comment:
this should be in the same level as the other describes instead of been inside the `Scan qrcode` block

Reply:
Does it make sense if i just break them apart ?

```     
     Expect.describe('Scan qrcode', function () {
      testScanQRCode(setupAndEnrollOktaTotp, setupOktaTotp, resTotpEnrollSuccess, 
       resAllFactors);
    });

    Expect.describe('Scan qrcode on Idx Pipeline', function () {
      testScanQRCode(setupAndEnrollOktaTotpWithIdxPipeline, setupOktaTotpWithIdxPipeline, resFactorEnrollTotpEnrollSuccess, resFactorEnrollAllFactors);
    });

    Expect.describe('Manual setup', function () {
      testScanQRCodeManualSetup(setupAndEnrollOktaTotp, setupOktaTotp, resTotpEnrollSuccess, 'testStateToken');
    });

    Expect.describe('Manual setup on Idx Pipeline', function () {
      testScanQRCodeManualSetup(setupAndEnrollOktaTotpWithIdxPipeline, setupOktaTotpWithIdxPipeline, resFactorEnrollTotpEnrollSuccess, '01testStateToken');
    });

    Expect.describe('Pass code form', function () {
      testScanQRCodePassCodeForm(setupAndEnrollOktaTotp, setupOktaTotp, resTotpEnrollSuccess, 'testStateToken');
    });
    
    Expect.describe('Pass code form on Idx Pipeline', function () {
      testScanQRCodePassCodeForm(setupAndEnrollOktaTotpWithIdxPipeline, setupOktaTotpWithIdxPipeline, resFactorEnrollTotpEnrollSuccess, '01testStateToken');
    });
```

Reply:
oh, I didn't realize they all already all grouped inside `Scan qrcode`, that's why you have them all grouped the same way... we have another `manual setup` inside a different describe.
I think now that I look at this again, that what makes more sense is to group them 
```
Expect.describe('Scan qrcode', function () {
  testScanQRCode(setupAndEnrollOktaTotp, setupOktaTotp, resTotpEnrollSuccess, 
   resAllFactors);

  Expect.describe('Manual setup', function () {
    testScanQRCodeManualSetup(setupAndEnrollOktaTotp, setupOktaTotp, resTotpEnrollSuccess, 'testStateToken');
  });

  Expect.describe('Pass code form', function () {
    testScanQRCodePassCodeForm(setupAndEnrollOktaTotp, setupOktaTotp, resTotpEnrollSuccess, 'testStateToken');
  });
});

Expect.describe('Scan qrcode on Idx Pipeline', function () {
  testScanQRCode(setupAndEnrollOktaTotpWithIdxPipeline, setupOktaTotpWithIdxPipeline, resFactorEnrollTotpEnrollSuccess, resFactorEnrollAllFactors);

  Expect.describe('Manual setup on Idx Pipeline', function () {
    testScanQRCodeManualSetup(setupAndEnrollOktaTotpWithIdxPipeline, setupOktaTotpWithIdxPipeline, resFactorEnrollTotpEnrollSuccess, '01testStateToken');
  });

  Expect.describe('Pass code form on Idx Pipeline', function () {
    testScanQRCodePassCodeForm(setupAndEnrollOktaTotpWithIdxPipeline, setupOktaTotpWithIdxPipeline, resFactorEnrollTotpEnrollSuccess, '01testStateToken');
  });
});

```

Reply:
Its arranged into idx and non-idx set of describes now.